src/main/java/pilcrow/Task.java

```diff
@@ -0,0 +1,232 @@
+package pilcrow;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Abstraction to represent a task.
+ */
+public class Task {
+    protected String taskName;
+    protected Boolean isDone;
+
+    private Task(String taskName, Boolean isDone) {
+        this.taskName = taskName;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Factory method to create a Task.
+     * @param taskType String representing type of Task to be created. May be "todo", "deadline" or "event".
+     * @param taskName Name of the Task to be created.
+     * @param isDone Done status of the Task to be created.
+     * @return Task with specified fields.
+     */
+    public static Task createTask(String taskType, String taskName, Boolean isDone) {
+        Task task;
+        if (taskType.equals("todo")) {
+            task = new ToDo(taskName, isDone);
+        } else if (taskType.equals("deadline")) {
+            task = new Deadline(taskName, isDone);
+        } else if (taskType.equals("event")) {
+            task = new Event(taskName, isDone);
+        } else {
+            // If not any of the task subtypes
+            // Should never reach this point
+            task = new Task("", false);
+        }
+        return task;
+    }
+
+    /**
+     * Converted a String that has been stored in Storage into a Task.
+     * @param storedTask String representing the stored task in Storage.
+     * @return Converted Task
+     */
+    public static Task convertFromStoredTask(String storedTask) {
+        Task task;
+        Parser parser = new Parser(storedTask);
+        String commandWord = parser.getCommandWord();
+        String restOfCommand = parser.getRestOfCommand().substring(1);
+        Boolean isDone = parser.getRestOfCommand().charAt(0) == 'T';
+
+        task = Task.createTask(commandWord, restOfCommand, isDone);
+        return task;
+    }
+
+    /**
+     * Dummy method, overridden by methods in subclasses.
+     * @return Dummy task
+     */
+    public String convertToStoredTask() {
+        String storedTask = "";
+        return storedTask;
+    }
+
+    public Boolean containsString(String string) {
+        return this.taskName.contains(string);
+    }
+
+    public void setIsDone(Boolean isDone) {
```

Comment:
Perhaps this method name could be `setDone` instead of `setIsDon` as it would read more naturally!